lms/djangoapps/ccx/tests/test_views.py

```diff
@@ -182,6 +186,130 @@ def test_forbidden_user_access_coach_dashboard(self):
+            )] for _ in xrange(2))
+
+    @patch('ccx.views.render_to_response', intercept_renderer)
+    @patch('courseware.views.views.render_to_response', intercept_renderer)
+    @patch.dict('django.conf.settings.FEATURES', {'CUSTOM_COURSES_EDX': True})
+    def test_edit_schedule(self):
+        """
+        Get CCX schedule, modify it, save it.
+        """
+        self.make_coach()
+        ccx = self.make_ccx()
+        ccx_course_key = CCXLocator.from_course_locator(self.course.id, unicode(ccx.id))
+        self.client.login(username=self.coach.username, password="test")
+
+        url = reverse('ccx_coach_dashboard', kwargs={'course_id': ccx_course_key})
+        response = self.client.get(url)
+
+        schedule = json.loads(response.mako_context['schedule'])  # pylint: disable=no-member
+        self.assertEqual(len(schedule), 1)
+        self.assertEqual(schedule[0]['hidden'], False)
+
+        def unhide(unit):
+            """
+            Recursively unhide a unit and all of its children in the CCX
+            schedule.
+            """
+            unit['hidden'] = False
+            for child in unit.get('children', ()):
+                unhide(child)
+
+        unhide(schedule[0])
+
+        # edit schedule
+        date = datetime.datetime.now() - datetime.timedelta(days=5)
+        start = date.strftime("%Y-%m-%d %H:%M")
+        due = (date + datetime.timedelta(days=3)).strftime("%Y-%m-%d %H:%M")
+
+        schedule[0]['start'] = start
+        schedule[0]['children'][0]['start'] = start
+        schedule[0]['children'][0]['due'] = due
+        schedule[0]['children'][0]['children'][0]['start'] = start
+        schedule[0]['children'][0]['children'][0]['due'] = due
+
+        url = reverse('save_ccx', kwargs={'course_id': ccx_course_key})
+        response = self.client.post(url, json.dumps(schedule), content_type='application/json')
+
+        self.assertEqual(response.status_code, 200)
+        schedule = json.loads(response.content)['schedule']
+        self.assertEqual(schedule[0]['hidden'], False)
+        self.assertEqual(schedule[0]['start'], start)
+        self.assertEqual(schedule[0]['children'][0]['start'], start)
+        self.assertEqual(schedule[0]['children'][0]['due'], due)
+        self.assertEqual(schedule[0]['children'][0]['children'][0]['due'], due)
+        self.assertEqual(schedule[0]['children'][0]['children'][0]['start'], start)
+
+        self.assert_progress_summery(ccx_course_key, due)
+
+    def assert_progress_summery(self, ccx_course_key, due):
+        """
+        assert singnal and schedule update.
+        """
+        student = UserFactory.create(is_staff=False)
+        CourseEnrollment.enroll(student, ccx_course_key)
+        self.assertTrue(
+            CourseEnrollment.objects.filter(course_id=ccx_course_key, user=student).exists()
+        )
+
+        # login as student
+        self.client.login(username=student.username, password=student.password)
```

Comment:
are you sure this is working? the password is usually encrypted and you cannot use it in this way